pkg/server/hostservices/agentstore/agentstore.go

```diff
@@ -0,0 +1,68 @@
+package agentstore
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/spiffe/spire/proto/spire/server/datastore"
+	"github.com/spiffe/spire/proto/spire/server/hostservices"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Deps struct {
+	// DataStore is used to retrieve the lastest bundle. It MUST be set.
+	DataStore datastore.DataStore
+}
+
+type AgentStore struct {
+	mu   sync.RWMutex
+	deps *Deps
+}
+
+func New() *AgentStore {
+	return &AgentStore{}
+}
+
+func (s *AgentStore) SetDeps(deps Deps) error {
```

Comment:
Couldn't help but notice that this function is exported but getDeps is not. Is that intentional?

Reply:
Oops, figured this one out later in the review... /delete